src/messaging/ExchangeContext.cpp

```diff
@@ -336,7 +364,6 @@ ExchangeContext::ExchangeContext(ExchangeManager * em, uint16_t ExchangeId, cons
 ExchangeContext::~ExchangeContext()
 {
     VerifyOrDie(mExchangeMgr != nullptr && GetReferenceCount() == 0);
-    VerifyOrDie(!IsAckPending());
```

Comment:
Why is this being removed?  We should not be ending up in a state where on destruction there is a pending ack; that means someone failed to flush acks correctly...

This is a must-fix.

Reply:
See also https://github.com/project-chip/connectedhomeip/pull/21846#discussion_r944548034